sdk/core/azure-core/src/main/java/com/azure/core/util/paging/ContinuablePagedByIteratorBase.java

```diff
@@ -100,6 +116,36 @@ synchronized void requestPage() {
         this.done = done || (!receivedPages.get() && !isNextAvailable());
     }
 
+    synchronized void requestPageSync() {
```

Comment:
There's a lot of common code in `requestPage` and `requestPageSync`. We can reduce duplication by extracting the common code into a separate method.